src/solver/expressions/visitors/CompareVisitor.cpp

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2007-2024, RTE (https://www.rte-france.com)
+ * See AUTHORS.txt
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of Antares-Simulator,
+ * Adequacy and Performance assessment for interconnected energy networks.
+ *
+ * Antares_Simulator is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public Licence 2.0 as published by
+ * the Mozilla Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Antares_Simulator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public Licence 2.0 for more details.
+ *
+ * You should have received a copy of the Mozilla Public Licence 2.0
+ * along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+ */
+#include <antares/solver/expressions/nodes/ExpressionsNodes.h>
+#include <antares/solver/expressions/visitors/CompareVisitor.h>
+
+template<class T, class V>
+static bool compareBinaryNode(V& visitor, const T& node, const Antares::Solver::Nodes::Node& other)
+{
+    if (const T* other_node = dynamic_cast<const T*>(&other))
+    {
+        bool left = visitor.dispatch(*node.left(), *other_node->left());
+        bool right = visitor.dispatch(*node.right(), *other_node->right());
+        return left && right;
+    }
+    return false;
+}
+
+template<class T>
+static bool compareGetValue(const T& node, const Antares::Solver::Nodes::Node& other)
+{
+    if (const T* other_node = dynamic_cast<const T*>(&other))
+    {
+        return node.value() == other_node->value();
+    }
+    return false;
+}
+
+template<class T>
+static bool compareEqualOperator(const T& node, const Antares::Solver::Nodes::Node& other)
```

Comment:
**compareEqualOperator** : 
if suggestions on **compareBinaryNode** work (not guaranteed) and are suitable, we could try them on **compareEqualOperator** and **compareGetValue**